src/cdk-experimental/popover-edit/edit-ref.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Subject} from 'rxjs';
+import {first} from 'rxjs/operators';
+import {Injectable, OnDestroy, Self} from '@angular/core';
+import {ControlContainer} from '@angular/forms';
+import {EditEvents} from './edit-events';
+
+/**
+ * Service used for communication between the form within the edit lens and the
+ * table that launched it. Provided by CdkPopoverEditControl within the lens.
+ */
+@Injectable()
+export class EditRef implements OnDestroy {
+  /** Emits the final value of this edit instance before closing. */
+  private readonly _finalValueSubject = new Subject<any>();
+  readonly finalValue = this._finalValueSubject.asObservable();
+
+  /** The value to set the form back to on revert. */
+  private _revertFormValue: any;
+
+  constructor(
+      @Self() private readonly _form: ControlContainer,
+      private readonly _editEvents: EditEvents, ) {}
+
+  /**
+   * Called by the host directive's OnInit hook. Reads the initial state of the
+   * form and overrides it with persisted state from previous openings, if
+   * applicable.
+   */
+  init(previousFormValue: any) {
+    // Wait for either the first value to be set, then override it with
+    // the previously entered value, if any.
+    this._form.valueChanges!.pipe(first()).subscribe(() => {
```

Comment:
We usually use `take(1)` instead of `first` since `first` will error if the stream never emits